apps/payments/wise.py

```diff
@@ -203,53 +203,57 @@ def wise_business_profile():
     return id
 
 
+def _retrieve_detail(details, requested_type):
+    """Helper method to retrieve content from attribute-value details recordsets"""
+    for detail in details:
+        if detail.type == requested_type:
+            return detail.body
+    raise AttributeError(f"Failed to find requested {requested_type} attribute in account details")
+
+
 def wise_retrieve_accounts(profile_id):
-    # Wise creates the concept of a multi-currency account by calling normal
-    # bank accounts "balances". As far as we're concerned, "balances" are bank
-    # accounts, as that's what people will be sending money to.
-    for account in wise.balances.list(profile_id=profile_id):
-        try:
-            if not account.bankDetails:
-                continue
-            if not account.bankDetails.bankAddress:
+    for account in wise.account_details.list(profile_id=profile_id):
+        account_holder = bank_name = bank_address = sort_code = account_number = swift = iban = None
+
+        if account.currency.code == "GBP":
+            for receive_options in account.receiveOptions:
+                details = receive_options.details
+                account_holder = _retrieve_detail(details, "ACCOUNT_HOLDER")
+                bank_info = _retrieve_detail(details, "BANK_NAME_AND_ADDRESS")
+
+                if receive_options.type == "LOCAL":
+                    sort_code = _retrieve_detail(details, "BANK_CODE").replace("-", "")
+                    account_number = _retrieve_detail(details, "ACCOUNT_NUMBER")
+
+                elif receive_options.type == "INTERNATIONAL":
+                    swift = _retrieve_detail(details, "SWIFT_CODE")
+                    iban = _retrieve_detail(details, "IBAN")
+
+            if not bank_info:
                 continue
-        except AttributeError:
-            continue
 
-        address = ", ".join(
-            [
-                account.bankDetails.bankAddress.addressFirstLine,
-                account.bankDetails.bankAddress.city + " " + (account.bankDetails.bankAddress.postCode or ""),
-                account.bankDetails.bankAddress.country,
-            ]
-        )
+            bank_name, _, bank_address = bank_info.partition("\n")
```

Comment:
This feels like it might come back to bite us at some point - perhaps we should remove the institution column from BankAccount and just throw this all in the address field.

Reply:
Yep, this does seem fragile.  However: it does seem useful in the admin UI (and perhaps elsewhere?) to have a small descriptive text field with the institution name alone.

I'll think about it some more.  Another option could be to add validation logic so that we detect any likely breakage early.